install.sh

```diff
@@ -0,0 +1,472 @@
+  version="$(curl -fsSLI -o /dev/null -w "%{url_effective}" https://github.com/cdr/code-server/releases/latest)"
+  version="${version#https://github.com/cdr/code-server/releases/tag/v}"
+  echo "$version"
+}
+
+echo_static_postinstall() {
+  echo
+  cat << EOF
+Static release has been installed into $ARCHIVE_INSTALL_PREFIX/lib/code-server-$VERSION
+Please extend your path to use code-server:
+  PATH="$ARCHIVE_INSTALL_PREFIX/bin:\$PATH"
+Then you can run:
+  code-server
+EOF
+}
+
+echo_systemd_postinstall() {
+  echo
+  cat << EOF
+To have systemd start code-server now and restart on boot:
+  systemctl --user enable --now code-server
+Or, if you don't want/need a background service you can run:
+  code-server
+EOF
+}
+
+main() {
+  if [ "${TRACE-}" ]; then
+    set -x
+  fi
+
+  unset \
+    DRY_RUN \
+    METHOD \
+    ARCHIVE_INSTALL_PREFIX \
+    SKIP_ECHO \
+    VERSION \
+    OPTIONAL
+
+  while [ "$#" -gt 0 ]; do
+    case "$1" in
+    --dry-run)
+      DRY_RUN=1
+      ;;
+    --method)
+      METHOD="$(parse_arg "$@")"
+      shift
+      ;;
+    --method=*)
+      METHOD="$(parse_arg "$@")"
+      ;;
+    --prefix)
+      ARCHIVE_INSTALL_PREFIX="$(parse_arg "$@")"
+      shift
+      ;;
+    --prefix=*)
+      ARCHIVE_INSTALL_PREFIX="$(parse_arg "$@")"
+      ;;
+    --version)
+      VERSION="$(parse_arg "$@")"
+      shift
+      ;;
+    --version=*)
+      VERSION="$(parse_arg "$@")"
+      ;;
+    -h | --h | -help | --help)
+      usage
+      exit 0
+      ;;
+    *)
+      echoerr "Unknown flag $1"
+      echoerr "Run with --help to see usage."
+      exit 1
+      ;;
+    esac
+
+    shift
+  done
+
+  VERSION="${VERSION-$(echo_latest_version)}"
+  METHOD="${METHOD-detect}"
+  if [ "$METHOD" != detect ] && [ "$METHOD" != archive ]; then
+    echoerr "Unknown install method \"$METHOD\""
+    echoerr "Run with --help to see usage."
+    exit 1
+  fi
+  ARCHIVE_INSTALL_PREFIX="${ARCHIVE_INSTALL_PREFIX-$HOME/.local}"
+
+  OS="$(os)"
+  if [ ! "$OS" ]; then
+    echoerr "Unsupported OS $(uname)."
+    exit 1
+  fi
+
+  distro_name
+
+  ARCH="$(arch)"
+  if [ ! "$ARCH" ]; then
+    if [ "$METHOD" = archive ]; then
+      echoerr "No static releases available for the architecture $(uname -m)."
+      echoerr 'Please rerun without the "--method archive" flag to install from npm.'
+      exit 1
+    fi
+    echo "No precompiled releases for $(uname -m)."
+    install_npm
+    return
+  fi
+
+  CACHE_DIR="$(echo_cache_dir)"
+  mkdir -p "$CACHE_DIR"
+
+  if [ "$METHOD" = archive ]; then
+    install_archive
+    return
+  fi
+
+  case "$(distro)" in
+  macos)
+    install_macos
+    ;;
+  ubuntu | debian | raspbian)
+    install_deb
+    ;;
+  centos | fedora | rhel | opensuse)
+    install_rpm
+    ;;
+  arch)
+    install_aur
+    ;;
+  *)
+    echo "Unsupported package manager."
+    install_archive
+    ;;
+  esac
+}
+
+parse_arg() {
+  case "$1" in
+  *=*)
+    opt="${1#=*}"
```

Comment:
I think this should be `%` instead of `#`.

Reply:
Ohh yea, nice catch. It's funny it works because the only time `opt` is printed is when there is no argument provided so it reads normally.

Reply:
lol yup I noticed the same thing, the only way to trigger is `--arg=` then it says `--arg= requires an argument`.